### PR TITLE
feat(content-sharing): Add config for email button

### PR DESCRIPTION
**Pull request by sahiga**
Add a configuration object that hides the email button in the USM.

An example of the ContentSharing UI Element without the email button:
content-sharing-email-button-hidden

The default USM:
regular-usm

### CODE REVIEW

**Comment by booc0mtaco** [comment]
@sahiga will this also act on sending emails for invite collaborations?

**Comment by sahiga** [comment]
@booc0mtaco Good question! Collaborators can still be invited via the "Invite People" form. This change just affects the "Email Shared Link" form, since we don't have an API for that feature yet.

**Comment by booc0mtaco** [comment]
@sahiga in that case, i am not sure it he key name and associated comment documenting it are specific enough. `SharingOptions` seems like it would (eventually) affect both. if it is meant specifically for disabling the email option for shared links, we may use a name that calls that out more deliberately.

**Comment by sahiga** [comment]
@booc0mtaco I updated the variable and type names. Let me know if these are clear enough, or if you have other thoughts.

File: src/features/unified-share-modal/flowTypes.js [hunk 1]
```diff
@@ -264,6 +264,11 @@ type EmailFormTypes = {
     sendSharedLinkError: React.Node,
 };
 
+export type USMConfig = {
+    /** Whether the "Email Shared Link" button and form should be rendered in the USM/USF */
+    showEmailSharedLinkForm: boolean,
+};
+
```
Comment (manishPh) [inline src/features/unified-share-modal/flowTypes.js:271]: question: Do we expect more settings to get added in future to this `USMConfig` object?

Comment (sahiga) [reply src/features/unified-share-modal/flowTypes.js:271]: Good question @manishPh. It's definitely a possibility, so I wanted to make the configuration object extensible.

**Pull request merged**